test/integration/client-side-operations-timeout/client_side_operations_timeout.prose.test.ts

```diff
@@ -192,6 +229,28 @@ describe('CSOT spec prose tests', function () {
        *   - Expect this to fail with a timeout error.
        * 1. Verify that an `insert` command was executed against to `keyvault.datakeys` as part of the `createDataKey` call.
        */
+
+      it('times out due to timeoutMS', clientEncryptionMetadata, async function () {
+        await internalClient
+          .db()
+          .admin()
+          .command({
+            configureFailPoint: 'failCommand',
+            mode: {
+              times: 1
+            },
+            data: {
+              failCommands: ['insert'],
+              blockConnection: true,
+              blockTimeMS: 150
+            }
+          } as FailPoint);
+        const commandStarted: CommandStartedEvent[] = [];
+        keyVaultClient.on('commandStarted', ev => commandStarted.push(ev));
+        const err = await clientEncryption.createDataKey('local').catch(e => e);
+        expect(err).to.be.instanceOf(MongoOperationTimeoutError);
+        expect(commandStarted[0]).to.containSubset({ commandName: 'insert' });
```

Comment:
The prose test says to verify that we executed an insert against the keyvault.datakeys, you can check that it executed against the right collection by checking `commandStarted[0].command.insert`